### PR TITLE
Code cleanup and optimization.

### PR DESCRIPTION
**Pull request by sudhirparmar123**
Below points have been fixed,

- Session   is being checked on all the controller which can moved to a common service

- URLs are   hardcoded, need to move these in a constant file

- Remove unnecessary line breaks and   space in the code

- Remove  unused files and  variables

- MongoDB   schema is not getting used in the employee route on the server

- Token is not sent in the header to check the user   authentication on rest APIs and implementation of   token check

- Confirmation   messages and other text is hardcoded in controller

- In angular app service, we have multiple methods per URL, this code can be   reduced to a common method with type and url   as parameter

- Remove Jquery code as much as possible

- Whole  employee object is being sent in the remove resume api, changed to send only userID

Thanks

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -57,3 +57,35 @@ resumeApp.config(function($stateProvider, $urlRouterProvider,$locationProvider)
...
+            currentUrl = toState.url;
+
+            if(currentUrl=='/login' && localStorage.getItem('token')!==null){
+                event.preventDefault();
+                $location.url('home');
+            } else if(authenticatedArea.indexOf(currentUrl) > -1) {
+                if(localStorage.getItem('token')===null) {
+                    event.preventDefault();
+                    $location.url('login')
+                    //$state.go('login');
```
Comment (csinghal) [inline app/app.js]: Delete commented code

File: app/config.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+//constants
+resumeApp.constant('SERVER_URL','http://172.26.35.76:8001/');
```
Comment (csinghal) [inline app/config.js]: try to make it dynamic

File: app/controllers/loginCtrl.js [hunk 3]
```diff
@@ -1,24 +1,23 @@
 resumeApp.controller('loginCtrl', function ($scope, toastService, $location, dashboardService, $state) {
     $scope.usertoken = "";
     $scope.loginBusy = false;
-    if (localStorage.getItem('token')) {
-        window.location.href = '/#/home';
-    }
+    $scope.showLoginBtn = true;
+    $scope.showLogoutBtn = true;
```
Comment (csinghal) [inline app/controllers/loginCtrl.js]: $scope = {
...$scope,
loginBusy: false
}

File: app/controllers/tabCtrl.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-resumeApp.controller('tabCtrl', function ($scope, toastService, utilService, $location, dashboardService) {
+resumeApp.controller('tabCtrl', function ($scope, toastService, utilService, $location) {
   $("#loginBtn").show();
```
Comment (csinghal) [inline app/controllers/tabCtrl.js]: try to remove jQuery

File: app/app.js (hunk 1 above)
Comment (sudhirparmar123) [reply app/app.js]: Fixed

File: app/config.js (hunk 2 above)
Comment (sudhirparmar123) [reply app/config.js]: Fixed

File: app/controllers/loginCtrl.js (hunk 3 above)
Comment (sudhirparmar123) [reply app/controllers/loginCtrl.js]: I tried this code, but it was showing error.

**Pull request merged**